__tests__/app.test.js

```diff
@@ -172,4 +165,75 @@ describe('/api/articles/:article_id/comments', () => {
     });
 });
 
+describe("Post api/articles/id/comment ", () => {
+    test("returns posted comment if a valid username is provided", () => {
+      const input = { username: "icellusedkars", body: "The username thing had me stumped!" };
+
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(input)
+        .expect(201)
+        .then(({ body : {comment}}) => {
+          expect(comment).toBeInstanceOf(Object);
+          expect(typeof comment.comment_id).toBe("number");
+          expect(comment.body).toBe("The username thing had me stumped!");
+          expect(comment.article_id).toBe(1);
+          expect(comment.votes).toBe(0);
+          expect(typeof comment.created_at).toBe("string");
+        });
+    });
+    test("returns posted comment when extra values given", () => {
+        const input = { username: "icellusedkars", body: "The username thing had me stumped!", extra: "unnecessary information" };
+  
+        return request(app)
+          .post("/api/articles/1/comments")
+          .send(input)
+          .expect(201)
+          .then(({ body : {comment}}) => {
+            expect(comment).toBeInstanceOf(Object);
+            expect(typeof comment.comment_id).toBe("number");
+            expect(comment.body).toBe("The username thing had me stumped!");
+            expect(comment.article_id).toBe(1);
+            expect(comment.votes).toBe(0);
+            expect(typeof comment.created_at).toBe("string");
+          });
+      });
+    test("returns 404 if article doesn't exist as it can't be found", () => {
+        const input = { username: "icellusedkars", body: "The username thing had me stumped!" };
+  
+        return request(app)
+          .post("/api/articles/999/comments")
+          .send(input)
+          .expect(404)
+          .then(({body}) => {
+            expect(body.msg).toBe('Not Found')
+          })
+    });
+    test("returns 400 if insufficient input is given - no username", () => {
+        const input = { body: "The username thing had me stumped!" };
+  
+        return request(app)
+          .post("/api/articles/1/comments")
+          .send(input)
+          .expect(400)
+          .then(({body}) => {
+            expect(body.msg).toBe('Bad Request')
+          })
+    });
+    test("returns 400 if insufficient input is given - no body", () => {
+        const input = { username: "icellusedkars" };
+  
+        return request(app)
+          .post("/api/articles/1/comments")
+          .send(input)
+          .expect(400)
+          .then(({body}) => {
+            expect(body.msg).toBe('Bad Request')
+          })
+    });
+});
+
```

Comment:
Love your testing for this, very thorough and you've accounted for any unnecessary properties too 👍